### PR TITLE
fix: tooltip unit test

### PR DESCRIPTION
**Pull request by dorianmiller**
#### What do these changes do/fix?

Because of overnight, unit tests were broken. This is the fix to define DOMRect 

#### How do you test/verify these changes?

Run unit test. On Mac M1, consider running this to build canvas:
1. nvm use && nvm install
2. brew install pkg-config cairo pango libpng

#### Have you documented your changes (if necessary)?
NA

#### Are there any breaking changes included in this pull request?
No

### CODE REVIEW

File: packages/discovery-react-components/src/components/DocumentPreview/__tests__/DocumentPreview.test.tsx [hunk 1]
```diff
@@ -210,7 +210,7 @@ describe('DocumentPreview', () => {
     });
 
     it(
-      'should render pdf',
+      'should render pdf (flaky, try to rerun)',
```
Comment (dorianmiller) [inline packages/discovery-react-components/src/components/DocumentPreview/__tests__/DocumentPreview.test.tsx:213]: I changed the title as I found 1/5 runs of yarn test failed

File: packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/__tests__/PdfViewer.test.tsx [hunk 2]
```diff
@@ -4,9 +4,12 @@ import { render, waitFor, screen } from '@testing-library/react';
 import 'web-streams-polyfill/es2018';
 import PdfViewer from '../PdfViewer';
 import { document as doc } from 'components/DocumentPreview/__fixtures__/Art Effects.pdf';
+import { setupBeforeRenderPdf } from 'setupTestsUtil';
 
 describe('PdfViewer', () => {
   it('renders PDF document', async () => {
+    // By experiment, it seems that setupBeforeRenderPdf is needed
+    setupBeforeRenderPdf();
```
Comment (dorianmiller) [inline packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/__tests__/PdfViewer.test.tsx:12]: By experiment, it seems that setupBeforeRenderPdf is needed.

I welcome suggestions if I missed a place to call `removeDOMRect()`

**Review by noah-eigenfeld — APPROVED** [review]

File: packages/discovery-react-components/src/setupTestsUtil.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export function removeDOMRect() {
+  delete globalThis.DOMRect;
+}
+
+// Ensure DOMRect is not defined for sake of PDF
+export function setupBeforeRenderPdf() {
+  removeDOMRect();
+}
```
Comment (jhpedemonte) [inline packages/discovery-react-components/src/setupTestsUtil.tsx:43]: I don't like this. It's not evident from the name of the function that it's just unsetting the DOMRect mock.

File: packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/__tests__/PdfViewer.test.tsx (hunk 2 above)
Comment (jhpedemonte) [reply packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/__tests__/PdfViewer.test.tsx:12]: Why? If it's removed at the end of previous tests, then it shouldn't be necessary here.

Comment (dorianmiller) [reply packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/__tests__/PdfViewer.test.tsx:12]: More experimenting, found it is not needed.

File: packages/discovery-react-components/src/components/DocumentPreview/__tests__/DocumentPreview.test.tsx (hunk 1 above)
Comment (dorianmiller) [reply packages/discovery-react-components/src/components/DocumentPreview/__tests__/DocumentPreview.test.tsx:213]: removed. Found to be more predictable.

File: packages/discovery-react-components/src/components/CIDocument/components/CIDocument/__tests__/CIDocument.spec.tsx [hunk 4]
```diff
@@ -26,12 +27,16 @@ describe('<CIDocument />', () => {
       queryByTestId: BoundFunction<QueryByText>;
 
     beforeEach(async () => {
+      defineDOMRect();
```
[the remark this answers is not in the document]
Comment (dorianmiller) [reply packages/discovery-react-components/src/components/CIDocument/components/CIDocument/__tests__/CIDocument.spec.tsx:30]: Learned more and it is `beforeAll` and `afterAll`. Implemented.

File: packages/discovery-react-components/src/setupTestsUtil.tsx (hunk 3 above)
Comment (dorianmiller) [reply packages/discovery-react-components/src/setupTestsUtil.tsx:43]: removed.

File: packages/discovery-react-components/src/components/CIDocument/components/CIDocument/__tests__/CIDocument.spec.tsx [hunk 5]
```diff
@@ -26,12 +27,16 @@ describe('<CIDocument />', () => {
...
       ({ getByTestId, getByText, findByText, findByTestId, queryByTestId } = render(
         <CIDocument document={invoice} overrideDocWidth={400} overrideDocHeight={600} />
       ));
       // wait for component to finish rendering (prevent "act" warning)
       await screen.findByText('invoice.pdf');
     });
+    afterEach(() => {
```
[the remark this answers is not in the document]
Comment (dorianmiller) [reply packages/discovery-react-components/src/components/CIDocument/components/CIDocument/__tests__/CIDocument.spec.tsx]: This particular call is removed. Concept is noted.

**Pull request merged**